source/base/function_spherical.cc

```diff
@@ -225,7 +225,7 @@ namespace Functions
   template <int dim>
   SymmetricTensor<2,dim>
   Spherical<dim>::hessian (const Point<dim> &p_,
-                           const unsigned int component) const
+                           const unsigned int /* component */) const
```

Comment:
Oh, this is a bug -- the component should actually be passed to `sgradient()` and `shessian()` below (line 232/233). Do you agree, @davydden ?